doc/modules/grid_search.rst

```diff
@@ -150,6 +154,217 @@ increasing ``n_iter`` will always lead to a finer search.
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching optimal parameters with successive halving
+====================================================
+
+Scikit-learn also provides the :class:`GridHalvingSearchCV` and
+:class:`RandomHalvingSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving is an iterative selection process where all candidates are evaluated
+with a small amount of resources at the first iteration. Only some of
+these candidates are selected for the next iteration, which will be
+allocated more resources. What defines a resource is typically the number of
+samples to train on, or the number of trees for a gradient boosting /
```

Comment:
number of iterations in iterative algorithms like gradient boosting? I mean for random forests it's not iterations but for linear models and neural nets it could be.